src/util.ts

```diff
@@ -14,21 +22,34 @@ export const EXTENSIONS = {
   map: 'suma',
 };
 
-let _superJson: SuperJson;
+let superJson: SuperJsonDocument;
 
-export function loadSuperJson(): SuperJson {
-  if (!_superJson) {
-    _superJson = SuperJson.loadSync().unwrap();
+export async function loadSuperJson(options?: {
+  fileSystem?: IFileSystem;
+}): Promise<{ document: SuperJsonDocument; path: string }> {
+  const superJsonPath = await detectSuperJson(
```

Comment:
You run this even when `superJson` is set. It is not necessary - just check in `superJson` is defined, return it if is is or detect path and load if it isn't.